### PR TITLE
Multitables 137409259

### PR DESCRIPTION
**Pull request by malecki**

### CODE REVIEW

File: source/includes/_endpoint-multitable.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#### POST
+
+POST a Shoji Entity to this catalog to create a new multitable definition. Entities must include a `name` and `template`; the template must contain a series of objects with keys `variable` and `query` and optionally `transform`. If omitted, `is_public` defaults to `false`. A successful POST yields a 201 response
+that will contain a Location header with the URL of the newly created multitable.
+
+All users with access to the dataset can create personal multitable definitions; however, only the current dataset editor can create public multitables (`is_public: true`) which everyone with access to the dataset can see. Attempting to create a public multitable when not the current dataset editor results in a 403 response.
+
+#### PATCH
+
+There are no elements of the catalog that can be changed via PATCH.
```
Comment (nealrichardson) [inline source/includes/_endpoint-multitable.md:40]: FWIW this is not consistent with other catalogs, which at least let you edit names.

File: source/includes/_endpoint-multitable.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+`/datasets/{dataset_id}/multitables/{multitable_id}/`
+
+#### GET
+
+```json
+{
+    "element": "shoji:entity",
+    "self": "datasets/123/multitables/fa1afe1/",
+    "views": {
+        "tabbook": "/datasets/123/multitables/7ab1e/tabbook/"
```
Comment (nealrichardson) [inline source/includes/_endpoint-multitable.md]: URL should match `self`

File: source/includes/_endpoint-multitable.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                "variable": "/datasets/123/variables/abc/"
+            }]
+        }, {
+            "variable": "/datasets/123/variables/def/",
+            "query": [{
+                "variable": "/datasets/123/variables/def/"
+            }]
+        }],
+        "is_public": false,
+        "id": "7ab1e"
```
Comment (nealrichardson) [inline source/includes/_endpoint-multitable.md:70]: Same.

File: source/includes/_endpoint-multitable.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+GET on this resource returns a Shoji Catalog with the list of Multitables that the
+current user can use on this Dataset.
+
+This index contains two kinds of multitables: those that belong to the dataset, denoted by the `is_public` tuple attribute; and those that belong to the current user. Personal multitables are those created by the authenticated user, and they cannot be accessed by other users. Dataset multitables are available to all users who are authorized to view the dataset.
+
+#### POST
+
+POST a Shoji Entity to this catalog to create a new multitable definition. Entities must include a `name` and `template`; the template must contain a series of objects with keys `variable` and `query` and optionally `transform`. If omitted, `is_public` defaults to `false`. A successful POST yields a 201 response
```
Comment (nealrichardson) [inline source/includes/_endpoint-multitable.md]: `template` needs specification somewhere.

Comment (nealrichardson) [reply source/includes/_endpoint-multitable.md]: This section should also link appropriately to the feature guide that talks about multitables; that feature guide should link back to here as appropriate and delete the duplicated content from there.

File: source/includes/_endpoint-multitable.md (hunk 1 above)
Comment (malecki) [reply source/includes/_endpoint-multitable.md:40]: https://www.pivotaltracker.com/story/show/138433131

File: source/includes/_feature-analyzing.md [hunk 5]
```diff
@@ -220,7 +214,22 @@ POST datasets/{id}/multitables/ HTTP/1.1
...
+Multi-table definitions mainly provide a `template` member that clients can use to construct a valid query with the variable(s) of interest.
+
+Crunch provides a separate catalog where you can define and manage these common sets of variables. Like most catalogs, you can [GET it](#multitables162) to see which multitables are defined.
+
+
+
+#### Template query
+
+A multitable is a set of queries that form groups of ‘columns’ for different later chosen ‘row’ variables.
+It is defined by a **name** and a **template**. At minimum the template must contain a `query` fragment:
```
Comment (nealrichardson) [inline source/includes/_feature-analyzing.md:225]: _It turns out_ that `variable` is being required by the validator: https://www.pivotaltracker.com/story/show/137627897/comments/161004753

File: source/includes/_feature-analyzing.md [hunk 6]
```diff
@@ -220,7 +214,22 @@ POST datasets/{id}/multitables/ HTTP/1.1
...
+this will be later inserted after some function of a row variable to form the [dimension](#dimensions)
+of a result. It is also required to indicate the variable involved in the query; each element of the template
```
Comment (nealrichardson) [inline source/includes/_feature-analyzing.md]: Oh. Then that previous sentence reads a little misleading.

File: source/includes/_endpoint-multitable.md (hunk 1 above)
Comment (nealrichardson) [reply source/includes/_endpoint-multitable.md:40]: 👍

File: source/includes/_feature-analyzing.md [hunk 7]
```diff
@@ -532,3 +431,132 @@ In a multitable, the `transform` is part of each dimension definition object in
     ]
 }
 ```
+
+
+### More complex multitable templates
+
+The template may contain in addition to variable references and their query arguments, an optional `transform`:
+To obtain their multiple output cubes, you `GET datasets/{id}/cube?query=<q>` where `<q>` is a ZCL object in JSON format (which must then be URI encoded for inclusion in the querystring). Use the "each" function to iterate over the overview variables’ `query`, producing one output cube for each one as "variable x". For example, to cross each of the above 3 variables against another variable "449b421":
```
Comment (nealrichardson) [inline source/includes/_feature-analyzing.md:432]: (1) is this valid? (2) link to cube docs.

**Pull request closed without merging**